action.yml

```diff
@@ -199,6 +199,73 @@ runs:
           cd /tmp/skaffold
           skaffold config set collect-metrics false
           skaffold deploy
+
+          if [ $? -eq 0 ]; then
+            skaffold render > rendered.yaml
+
+            container_images=""
+            if yq -e '.spec.template.spec.containers[]?.image' "rendered.yaml" > /dev/null 2>&1; then
+              # Extract the unique container images from the rendered chart
+              container_images=$(yq '.spec.template.spec.containers[].image' "rendered.yaml" | sort -u)
+            fi
+
+            purl=""
+            if [ -n "$container_images" ]; then
+              echo "Saving artifact information for chart $CHART_LOCATION"
+
+              helm_version=$CHART_VERSION
+              if [ -z "$helm_version" ]; then
+                if [ -f "$CHART_PATH" ] && [[ "$CHART_PATH" == *.tgz ]]; then
+                  artifactPath="$CHART_PATH"
+                else
+                  chartArtifactDir="/tmp/helm-chart-artifact"
+                  if helm package $CHART_PATH --destination "$chartArtifactDir" > /dev/null 2>&1; then
+                    artifactPath=$(ls -Art "$chartArtifactDir"/*.tgz | tail -n 1)
+                  fi
+                fi
+                if [ -f "$artifactPath" ]; then
+                  helm_version="sha256:$(sha256sum "$artifactPath" | awk '{print $1}')"
+                fi
+              fi
+  
+              if [[ "$CHART_LOCATION" =~ ^oci:// ]]; then
+                # OCI image format
+                sha256_digest=$(helm pull "$CHART_LOCATION" 2>&1 | grep 'Digest:' | sed 's/Digest: //' || echo "" )
```

Comment:
@mcritic I am thinking that now that we will be adding `digest` support, we don't need to use the purl format for a reliable match for helm charts. 
So lets use the format of `<CHART_LOCATION>:<CHART_VERSION>` for all remote paths. For local path, we can just use `<CHART_NAME>:<CHART_VERSION>`. 
CHART_NAME and CHART_VERSION are `name` and `version` in the `Chart.yaml`. 
Keep the logic you have for determining the sha256_digest value in all the cases. We will use it as the `digest` value later. 
Sorry for changing the design after you added all this logic. It took me a while to realize the dots were not connecting properly for some of these cases for artifact traceability without the digest.